src/main/java/com/ethercamp/harmony/web/controller/ContractsController.java

```diff
@@ -40,87 +44,106 @@
     public ActionStatus<ContractInfoDTO> uploadContractFiles(
             @PathVariable String address,
             @RequestParam MultipartFile[] contracts,
-            @RequestParam(required = false) String verifyRlp) {
+            @RequestParam(required = false) String verifyRlp) throws DisabledException {
 
         try {
             ContractInfoDTO contract = contractsService.uploadContract(lowerCase(address), contracts);
             log.info("Uploaded files for address: {}, contract name: {}" + address, contract.getName());
             return createSuccessStatus(contract);
+        } catch (DisabledException e) {
+            throw e;
         } catch (Exception e) {
             log.warn("Contract's source uploading error: ", e);
             return createErrorStatus(e.getMessage());
         }
     }
 
-    @RequestMapping(value = "/contracts/{address}/importFromExplorer", method = RequestMethod.POST)
-    public ActionStatus<Boolean> importContractDataFromExplorer(@PathVariable String address) {
+    @RequestMapping(value = "/{address}/importFromExplorer", method = RequestMethod.POST)
+    public ActionStatus<Boolean> importContractDataFromExplorer(@PathVariable String address) throws DisabledException {
         try {
             final boolean result = contractsService.importContractFromExplorer(address);
             return createSuccessStatus(result);
+        } catch (DisabledException e) {
+            throw e;
         } catch (Exception e) {
             return createErrorStatus(e.getMessage());
         }
     }
 
-    @RequestMapping(value = "/contracts/{address}/clearContractStorage", method = RequestMethod.POST)
-    public ActionStatus<Boolean> clearStorage(@PathVariable String address) {
+    @RequestMapping(value = "/{address}/clearContractStorage", method = RequestMethod.POST)
+    public ActionStatus<Boolean> clearStorage(@PathVariable String address) throws DisabledException {
         try {
             contractsService.clearContractStorage(address);
             return createSuccessStatus();
+        } catch (DisabledException e) {
+            throw e;
         } catch (Exception e) {
             return createErrorStatus(e.getMessage());
         }
     }
 
-    @RequestMapping("/contracts/indexStatus")
-    public ActionStatus<IndexStatusDTO> getIndexStatus() {
+    @RequestMapping("/indexStatus")
+    public ActionStatus<IndexStatusDTO> getIndexStatus() throws DisabledException {
         try {
             IndexStatusDTO result = contractsService.getIndexStatus();
             return createSuccessStatus(result);
+        } catch (DisabledException e) {
+            throw e;
         } catch (Exception e) {
             log.warn("Contract's index status error: ", e);
             return createErrorStatus(e.getMessage());
         }
     }
 
+    // Convert a predefined exception to an HTTP Status code
+    @ResponseStatus(value=HttpStatus.GONE, reason="Service is disabled")  // 410
```

Comment:
Not sure this is the best place for exception handlers. Don't we have a separate controller for that?

Reply:
It's per-controller type of handler.